src/main/scala/is/hail/annotations/aggregators/RegionValueVirtualDSAggregator.scala

```diff
@@ -0,0 +1,52 @@
+package is.hail.annotations.aggregators
+import is.hail.annotations.RegionValueBuilder
+import is.hail.annotations._
+import is.hail.expr.types.{TArray, TFloat64, TTuple}
+import org.apache.spark.sql.Row
+
+object RegionValueVirtualDSAggregator {
+  val typ = TArray(TTuple(Array(TFloat64(), TFloat64())))
+}
+
+class RegionValueVirtualDSAggregator(xmin: Double, xmax: Double, ymin: Double, ymax: Double) extends RegionValueAggregator {
+  private var boxes = Set.empty[(Int, Int)]
+
+  def seqOp(r: Region, offset: Long, missing: Boolean) {
+    if (!missing) {
+      val row = UnsafeRow.read(TTuple(Array(TFloat64(), TFloat64())), r, offset).asInstanceOf[UnsafeRow]
+      val x = row.getDouble(0)
+      val y = row.getDouble(1)
+      val i = ((x / ((xmax - xmin) / 10)).toInt, (y / ((ymax - ymin) / 10)).toInt)
+      boxes += i
+    }
+  }
+
+  def combOp(agg2: RegionValueAggregator) {
+    boxes = boxes ++ agg2.asInstanceOf[RegionValueVirtualDSAggregator].boxes
+  }
+
+  def result(rvb: RegionValueBuilder) {
+    rvb.startArray(boxes.size)
+    boxes.foreach(i => {
```

Comment:
You can write this:

```
  boxes.iterator.foreach { i =>
    val x = ...
  }
```

which is a little more concise because you only have one set of parens/braces.